File: src/model/item/Animal.java
package model.item;

import java.util.ArrayList;
import java.util.List;

//import component.PartyMemberDataSource;
import core.ConstantStore;

import model.Conditioned;
import model.Item;

@SuppressWarnings("serial")
public class Animal extends Item implements Conditioned {
	private double moveFactor;
	
	private boolean dead;
	
	public Animal(ItemType type) {
		super(type);
		this.moveFactor = Double.parseDouble(ConstantStore.get("ITEMS", type + "_MOVE_FACTOR"));
	}
	
	public List<Item> killForFood() {
		int numberOf = (int) (this.getWeight() / ItemType.MEAT.getWeight());
		List<Item> itemList = new ArrayList<Item>();
		for(int i = 0; i < (numberOf / 10); i++) {
			itemList.add(new Item(ItemType.MEAT));
		}
		return itemList;
	}
	
	public double getMoveFactor() {
		return moveFactor;
	}

	public void setDead(boolean dead) {
		this.dead = dead;
	}

//	@Override
	public boolean isDead() {
		return dead;
	}
}

File: web/src/model/Animal.ts
import { Item } from "./Item";
import { ItemType, getItemTypeWeight } from "./ItemType";
import { ConstantStore } from "../core/ConstantStore";

export class Animal extends Item {
  private moveFactor: number;
  private dead: boolean;

  constructor(type: ItemType) {
    super(type);
    const moveFactorStr = ConstantStore.get("ITEMS", `${type}_MOVE_FACTOR`);
    this.moveFactor = moveFactorStr ? parseFloat(moveFactorStr) : 0;
    this.dead = false;
  }

  killForFood(): Item[] {
    const meatWeight = getItemTypeWeight(ItemType.MEAT);
    const numberOf = Math.floor(this.getWeight() / meatWeight);
    const itemList: Item[] = [];
    for (let i = 0; i < Math.floor(numberOf / 10); i++) {
      itemList.push(new Item(ItemType.MEAT));
    }
    return itemList;
  }

  getMoveFactor(): number {
    return this.moveFactor;
  }

  setDead(dead: boolean): void {
    this.dead = dead;
  }

  isDead(): boolean {
    return this.dead;
  }
}
